### ISSUE TITLE
playFromUrl fails on iOS when URL is audio stream

### ISSUE THREAD

**Issue by jayvolr** [nstudio/nativescript-audio#99]
I'm trying to play audio from multiple URLs of live audio streams. It works just fine on android, but iOS is giving me trouble.

I've created a minimal nativescript app with steps in the README.md to easily reproduce this:

https://github.com/jayvolr/nativescript-audio-stream-error

**Comment by jayvolr** [comment]
To clarify, I'm not trying to play multiple streams at once, there's just multiple URLs which have this issue on iOS (anything live and not pre-recorded).

**Comment by jayvolr** [comment]
This answer on StackOverflow seems like it might be related: https://stackoverflow.com/a/34563570

It's not dealing with NativeScript, but the poster says "Use AVPlayer instead of AVAudioPlayer to play remote content. As per documentation AVAudioPlayer needs mp3 file to play Audio. AVAudioPlayer not provide support for [streaming](https://developer.apple.com/library/content/qa/qa1634/_index.html)."

nativescript-audio uses AVAudioPlayer, correct?

**Comment by bradmartin** [comment]
That is correct. See [here](https://github.com/nstudio/nativescript-audio/blob/master/src/ios/player.ts#L105) for the implementation. Could try switching the player implementation out with https://developer.apple.com/documentation/avfoundation/avplayer class. I don't know the time/work involved. Would depend if it would be necessary to switch out the entire audio player on iOS or if some internal work could be done to us AVPlayer just for remote content and leave the current class working on local files.

**Comment by janstoeckler** [comment]
This https://github.com/janstoeckler/nativescript-audio/tree/ios-avplayer-reroll is a "reroll" of sorts of the ios-avplayer against the current master w/ just the necessary changes to ios/player.ts, don't know if that's helpful or not, but it seems to work.

**Comment by bradmartin** [comment]
I have this done locally, just need to handle merging the changes and not causing too many issues/breaking changes.

**Comment by 7ammer** [comment]
Hello @janstoeckler,
I have tried pulling in your fork but unfortunately doing the below returns `null`. Android works fine though. 

```
import { TNSPlayer } from '/path/to/local/copy/of/janstoeckler/nativescript-audio/ios-avplayer-reroll';

let player = new TNSPlayer();
console.log(player); // returns null
```
I wondered if you knew what was wrong?

Thanks

----

nativescript version:5.4.2

**Comment by bradmartin** [comment]
this plugin will be the future work of playing audio files https://github.com/nstudio/nativescript-audio-player It uses AVPlayer, has no built in recording as splitting the plugins made more sense overall to us going forward. We haven't had time to tidy up the documentation or API on it but it should work very similar to this plugin as it was copy/paste with some tweaks and changes.

**Comment by 7ammer** [comment]
@bradmartin That's great! Thank you 👍 
I'll check it out and let you know if I run into anything.

Many thanks again! :)

**Comment by 7ammer** [comment]
Any updates? nstudio/nativescript-audio-player is looking a bit dead :L
